controller_interface/include/semantic_components/force_torque_sensor.hpp

```diff
@@ -0,0 +1,169 @@
+  : SemanticComponentInterface(name, 6)
+  {
+    // If 6D FTS use standard names
+    interface_names_.emplace_back(name_ + "/" + "force.x");
+    interface_names_.emplace_back(name_ + "/" + "force.y");
+    interface_names_.emplace_back(name_ + "/" + "force.z");
+    interface_names_.emplace_back(name_ + "/" + "torque.x");
+    interface_names_.emplace_back(name_ + "/" + "torque.y");
+    interface_names_.emplace_back(name_ + "/" + "torque.z");
+
+    // Set all interfaces existing
+    std::fill(existing_axes_.begin(), existing_axes_.end(), true);
+
+    // Set default force and torque values to NaN
+    std::fill(forces_.begin(), forces_.end(), std::numeric_limits<double>::quiet_NaN());
+    std::fill(torques_.begin(), torques_.end(), std::numeric_limits<double>::quiet_NaN());
+  }
+
+  /// Constructor for 6D FTS with custom interface names.
+  /**
+   * Constructor for 6D FTS with custom interface names or FTS with less then six measurement axes,
+   * e.g., 1D and 2D force load cells.
+   * For non existing axes interface is empty string, i.e., ("");
+   *
+   * The name should be in the following order:
+   *   force X, force Y, force Z, torque X, torque Y, torque Z.
+   */
+  ForceTorqueSensor(
+    const std::string & interface_force_x, const std::string & interface_force_y,
+    const std::string & interface_force_z, const std::string & interface_torque_x,
+    const std::string & interface_torque_y, const std::string & interface_torque_z
+  )
+  : SemanticComponentInterface("", 6)
+  {
+    auto check_and_add_interface =
+      [this](const std::string & interface_name, const int index)
+      {
+        if (!interface_name.empty()) {
+          interface_names_.emplace_back(interface_name);
+          existing_axes_[index] = true;
+        } else {
+          existing_axes_[index] = false;
+        }
+      };
+
+    check_and_add_interface(interface_force_x, 0);
+    check_and_add_interface(interface_force_y, 1);
+    check_and_add_interface(interface_force_z, 2);
+    check_and_add_interface(interface_torque_x, 3);
+    check_and_add_interface(interface_torque_y, 4);
+    check_and_add_interface(interface_torque_z, 5);
+
+    // Set default force and torque values to NaN
+    std::fill(forces_.begin(), forces_.end(), std::numeric_limits<double>::quiet_NaN());
+    std::fill(torques_.begin(), torques_.end(), std::numeric_limits<double>::quiet_NaN());
+  }
+
+  /// Return forces.
+  /**
+   * Return forces of a FTS.
+   *
+   * \return array of size 3 with force values.
+   */
+  std::array<double, 3> get_forces()
+  {
+    size_t interface_counter = 0;
+    for (size_t i = 0; i < 3; ++i) {
+      if (existing_axes_[i]) {
+        forces_[i] = state_interfaces_[interface_counter].get().get_value();
+        ++interface_counter;
+      }
+    }
+    return forces_;
+  }
+
+  /// Return torque.
+  /**
+   * Return torques of a FTS.
+   *
+   * \return array of size 3 with torque values.
+   */
+  std::array<double, 3> get_torques()
```

Comment:
@jordan-palacios is this ok from an RT perspective? only vectors were problematic right?

Reply:
@bmagyar Yeah, arrays are fine.